kernel/poll.c

```diff
@@ -214,10 +214,9 @@ static inline int register_events(struct k_poll_event *events,
 			poller->is_polling = false;
 		} else if (!just_check && poller->is_polling) {
 			int rc = register_event(&events[ii], poller);
```

Comment:
Now that i look at the implementation of `register_event`, see https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/poll.c#L121 It always returns 0.

I think the proper solution here would be:
1. change `static inline int register_event()` into `static inline void register_event()` and remove the `return 0` inside the function.
2. In this line (line 216) remove the variable rc, so you have here sth like:
```
register_event(&events[ii], poller);
events_registered += 1;
```

@nashif What do you think? Or should `register_event()` actually return a value?

Reply:
yes, that works.